third-party/linshare/src/test/java/org/apache/james/linshare/LinshareBlobExportMechanismTest.java

```diff
@@ -0,0 +1,116 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hamcrest.Matchers.containsString;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.HashBlobId;
+import org.apache.james.blob.export.api.BlobExportMechanism;
+import org.apache.james.blob.export.api.FileExtension;
+import org.apache.james.blob.memory.MemoryBlobStore;
+import org.apache.james.core.MailAddress;
+import org.apache.james.linshare.client.LinshareAPI;
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class LinshareBlobExportMechanismTest {
+    private static final String EXPLANATION = "Explanation about the file being shared";
+
+    @RegisterExtension
+    static LinshareExtension linshareExtension = new LinshareExtension();
+
+    private MemoryBlobStore blobStore;
+    private LinshareBlobExportMechanism testee;
+    private HashBlobId.Factory blobIdFactory;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        blobIdFactory = new HashBlobId.Factory();
+        blobStore = new MemoryBlobStore(blobIdFactory);
+
+        testee = new LinshareBlobExportMechanism(
+            linshareExtension.getAPIFor(USER_1),
+            blobStore);
+    }
+
+    @Test
+    void exportShouldShareTheDocumentViaLinShare() throws Exception {
+        BlobId blobId = blobStore.save("content".getBytes(StandardCharsets.UTF_8)).block();
+
+        testee.blobId(blobId)
+            .with(new MailAddress(USER_2.getUsername()))
+            .explanation(EXPLANATION)
+            .fileExtension(FileExtension.of("txt"))
+            .export();
+
+        LinshareAPI user2API = linshareExtension.getAPIFor(USER_2);
+
+        assertThat(user2API.receivedShares())
+            .hasSize(1)
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getDocument().getName()).endsWith(".txt"))
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getDocument().getName()).startsWith(blobId.asString()))
+            .allSatisfy(receivedShare -> assertThat(receivedShare.getSender().getMail()).isEqualTo(USER_1.getUsername()));
+    }
+
+    @Test
+    void exportShouldSendAnEmailToSharee() throws Exception {
+        BlobId blobId = blobStore.save("content".getBytes(StandardCharsets.UTF_8)).block();
+
+        testee.blobId(blobId)
+            .with(new MailAddress(USER_2.getUsername()))
+            .explanation(EXPLANATION)
+            .fileExtension(FileExtension.of("txt"))
+            .export();
+
+
+        Awaitility.waitAtMost(Duration.TEN_SECONDS)
+            .pollInterval(Duration.ONE_SECOND)
+            .untilAsserted(
+                () -> given(linshareExtension.getLinshare().fakeSmtpRequestSpecification())
+                    .get("/api/email")
```

Comment:
can we get mails from the sharee (USER_2), not all emails? Or assert the receiver is USER_2? It would be more precise

Reply:
Seems code of the fake smtp server taken is this : https://github.com/ambled/rest-smtp-sink
In `index.js` I don't see anything that allows you to get a mail from a specific user...
I will check though if we can assert that the receiver is USER_2